### PR TITLE
feat: only add existing app insights instrumentation key

### LINKED ISSUE #563: Only add Azure Application Insights Serilog sink if the instrumentation key is present
**Is your feature request related to a problem? Please describe.**
Currently, all our project templates adds an Azure Application Insights Serilog sink without checking the contents of the instrumentation key first. This could lead to startup failures due to only the logging infrastructure of the application.

**Describe the solution you'd like**
Check the contents of the instrumentation key first before adding the Azure Application Insights Serilog sink so that the project starts regardless of a valid logging infrastructure. Log a warning in that case so that the developer knows the logging infrastructure is missing something.

**Additional context**
FIrst mentioned in #550 by my good buddy @fgheysels

### PR DESCRIPTION
**Pull request by stijnmoreels**
Only add an Azure Application Insights instrumentation key if such value is present.

Closes #563

### CODE REVIEW

File: src/Arcus.Templates.AzureFunctions.Databricks.JobMetrics/Startup.cs [hunk 1]
```diff
@@ -20,39 +19,50 @@ public override void ConfigureAppConfiguration(IFunctionsConfigurationBuilder bu
...
                 .Enrich.WithVersion()
-                .WriteTo.Console()
-                .WriteTo.AzureApplicationInsights(instrumentationKey);
-
-            builder.Services.AddLogging(logging =>
+                .WriteTo.Console();
+            
+            IConfiguration appConfig = builder.GetContext().Configuration;
+            var instrumentationKey = appConfig.GetValue<string>("APPINSIGHTS_INSTRUMENTATIONKEY");
+            if (instrumentationKey != null)
```
Comment (fgheysels) [inline src/Arcus.Templates.AzureFunctions.Databricks.JobMetrics/Startup.cs:60]: Why not use String.IsNullOrWhitespace or something like that ?

File: src/Arcus.Templates.ServiceBus.Queue/Program.cs [hunk 2]
```diff
@@ -13,25 +13,19 @@ namespace Arcus.Templates.ServiceBus.Queue
 {
     public class Program
     {
-#if (ExcludeSerilog == false)
+#if Serilog
         #warning Make sure that the appsettings.json is updated with your Azure Application Insights instrumentation key.
         private const string ApplicationInsightsInstrumentationKeyName = "TELEMETRY_APPLICATIONINSIGHTS_INSTRUMENTATIONKEY";
```
Comment (fgheysels) [inline src/Arcus.Templates.ServiceBus.Queue/Program.cs:18]: We should align the name of this setting so that it also uses `APPINSIGHTS_INSTRUMENTATIONKEY`

File: src/Arcus.Templates.ServiceBus.Topic/Program.cs [hunk 3]
```diff
@@ -13,20 +13,14 @@ namespace Arcus.Templates.ServiceBus.Topic
 {
     public class Program
     {
-#if (ExcludeSerilog == false)
+#if Serilog
         #warning Make sure that the appsettings.json is updated with your Azure Application Insights instrumentation key.
         private const string ApplicationInsightsInstrumentationKeyName = "TELEMETRY_APPLICATIONINSIGHTS_INSTRUMENTATIONKEY";
```
Comment (fgheysels) [inline src/Arcus.Templates.ServiceBus.Topic/Program.cs:18]: Same here, the setting should be called `APPINSIGHTS_INSTRUMENTATIONKEY`

File: src/Arcus.Templates.ServiceBus.Queue/Program.cs (hunk 2 above)
Comment (stijnmoreels) [reply src/Arcus.Templates.ServiceBus.Queue/Program.cs:18]: Yes, I know, is still WIP (draft PR + 🚧).

File: src/Arcus.Templates.AzureFunctions.Databricks.JobMetrics/Startup.cs (hunk 1 above)
Comment (stijnmoreels) [reply src/Arcus.Templates.AzureFunctions.Databricks.JobMetrics/Startup.cs:60]: Will do.
Is still WIP (draft PR + 🚧).

File: src/Arcus.Templates.ServiceBus.Topic/Program.cs (hunk 3 above)
Comment (stijnmoreels) [reply src/Arcus.Templates.ServiceBus.Topic/Program.cs]: Done in https://github.com/arcus-azure/arcus.templates/pull/587/commits/844d1b8d5f71c36dc7c34c097bf47629a2088bd1

File: src/Arcus.Templates.ServiceBus.Queue/Program.cs (hunk 2 above)
Comment (stijnmoreels) [reply src/Arcus.Templates.ServiceBus.Queue/Program.cs]: Done in https://github.com/arcus-azure/arcus.templates/pull/587/commits/844d1b8d5f71c36dc7c34c097bf47629a2088bd1

File: src/Arcus.Templates.AzureFunctions.Databricks.JobMetrics/Startup.cs (hunk 1 above)
Comment (stijnmoreels) [reply src/Arcus.Templates.AzureFunctions.Databricks.JobMetrics/Startup.cs]: Done in https://github.com/arcus-azure/arcus.templates/pull/587/commits/844d1b8d5f71c36dc7c34c097bf47629a2088bd1

File: src/Arcus.Templates.AzureFunctions.Databricks.JobMetrics/Startup.cs [hunk 4]
```diff
@@ -20,39 +19,50 @@ public override void ConfigureAppConfiguration(IFunctionsConfigurationBuilder bu
...
             builder.ConfigureSecretStore(stores =>
             {
 //[#if DEBUG]
                 stores.AddConfiguration(config);
 //[#endif]
-
+                
                 stores.AddEnvironmentVariables();
-
+
```
Comment (fgheysels) [inline src/Arcus.Templates.AzureFunctions.Databricks.JobMetrics/Startup.cs:36]: Unnecessary whitespace :)

**Review by fgheysels — APPROVED** [review]

File: src/Arcus.Templates.AzureFunctions.Databricks.JobMetrics/Startup.cs (hunk 4 above)
Comment (stijnmoreels) [reply src/Arcus.Templates.AzureFunctions.Databricks.JobMetrics/Startup.cs:36]: Not exactly. The dotnet template sometimes removes the lines without characters, so the result is an all pushed together in a not clean code file. The whitespace makes sure that we don't have that.
